rust/connlib/snownet/src/node.rs

```diff
@@ -2346,14 +2358,32 @@ where
         control_flow
     }
 
-    fn force_handshake(
+    fn initiate_wg_session(
         &mut self,
         allocations: &mut BTreeMap<RId, Allocation>,
         transmits: &mut VecDeque<Transmit>,
         now: Instant,
     ) where
         RId: Copy,
     {
+        let Some(socket) = self.socket() else {
+            tracing::error!("Cannot initiate WG session without a socket");
+            return;
+        };
+
+        // If we have sent a handshake in the last 20s, don't bother making a new session.
```

Comment:
What if the handshake got lost? Would this negate any kind of retry mechanism?

Reply:
boringtun retries a handshake every 5s, up until to rekey-timeout which we set to 15s.